mlflow/store/tracking/dbmodels/models.py

```diff
@@ -1637,13 +1644,18 @@ def merge(self, new_record_dict: dict[str, Any]) -> None:
         """
         Merge new record data into this existing record.
 
-        Updates expectations and tags by merging new values with existing ones.
+        Updates outputs, expectations and tags by merging new values with existing ones.
         Preserves created_time and created_by from the original record.
 
         Args:
             new_record_dict: Dictionary containing new record data with optional
-                           'expectations' and 'tags' fields to merge.
+                           'outputs', 'expectations' and 'tags' fields to merge.
         """
+        if new_outputs := new_record_dict.get("outputs"):
```

Comment:
I don't think that we should be appending outputs content to the existing (if present) outputs. This should probably be a replace command instead of an append mutation.